quran/test_models.py

```diff
@@ -0,0 +1,105 @@
+import shutil
+import tempfile
+from django.core import exceptions
+from django.conf import settings
+from django.test import TestCase
+from django.core.files import File
+from .models import Ayah, Reciter, Recitation
+
+
+class ReciterTestCase(TestCase):
+    """Tests for Reciter model"""
+
+    def setUp(self):
+        # create the reciter object
+        self.mishari = Reciter.objects.create(
+            name="Mishari Rashid al-ʿAfasi")
+
+    def test_reciter_string_representation(self):
+        self.assertEqual(str(self.mishari), 'Mishari Rashid al-ʿAfasi')
+
+    def test_reciter_string_representation_with_style(self):
+        self.mishari.style = 'murattal'
+        self.assertEqual(
+            str(self.mishari), 'Mishari Rashid al-ʿAfasi-murattal')
+
+    def test_reciter_string_representation_with_style_and_bitrate(self):
+        self.mishari.style = 'murattal'
+        self.mishari.bitrate = 128
+        self.assertEqual(
+            str(self.mishari), 'Mishari Rashid al-ʿAfasi-murattal-128kb/s')
+
+    def test_reciter_slug(self):
+        self.assertEqual(self.mishari.slug, 'mishari-rashid-al-afasi')
+
+
+class RecitationTestCase(TestCase):
```

Comment:
You should write a separate test case for your `SegmentsField`. It's independent of your model in fact. Try to read some Django code and see how do they test their fields.